rotkehlchen/history/events.py

```diff
@@ -365,17 +347,34 @@ def fail_history_cb(error_msg: str) -> None:
         ethereum.transactions_decoder.get_and_decode_undecoded_transactions(limit=None)
         step = self._increase_progress(step, total_steps)
 
-        # Include all external trades and trades from external exchanges
-        for location in EXTERNAL_LOCATION:
-            self.processing_state_name = f'Querying {location} trades history'
-            with self.db.conn.read_ctx() as cursor:
-                external_trades = self.db.get_trades(
-                    cursor,
-                    filter_query=TradesFilterQuery.make(location=location),
-                    has_premium=True,  # we need all trades for accounting -- limit happens later
-                )
-            history.extend(external_trades)
-            step = self._increase_progress(step, total_steps)
+        # Include all trades
+        self.processing_state_name = 'Querying trades history'
+        with self.db.conn.read_ctx() as cursor:
+            trades = self.db.get_trades(
+                cursor,
+                filter_query=TradesFilterQuery.make(),
+                has_premium=True,  # we need all trades for accounting -- limit happens later
+            )
+        history.extend(trades)
+        step = self._increase_progress(step, total_steps)
+
+        # Include all asset movements
+        self.processing_state_name = 'Querying asset movements history'
+        with self.db.conn.read_ctx() as cursor:
+            asset_movements = self.db.get_asset_movements(
+                cursor,
+                filter_query=AssetMovementsFilterQuery.make(),
+                has_premium=True,  # we need all trades for accounting -- limit happens later
+            )
+        history.extend(asset_movements)
+        step = self._increase_progress(step, total_steps)
+
+        # Include all margin positions
+        self.processing_state_name = 'Querying trades history'
+        with self.db.conn.read_ctx() as cursor:
+            margin_positions = self.db.get_margin_positions(cursor)
```

Comment:
Can't you just put everything under the same with statement. I believe is faster than opening and closing the cursor each time